homeassistant/components/wolflink/config_flow.py

```diff
@@ -57,16 +58,19 @@ async def async_step_user(
             step_id="user", data_schema=USER_SCHEMA, errors=errors
         )
 
-    async def async_step_device(self, user_input=None):
+    async def async_step_device(
+        self, user_input: dict[str, str] | None = None
+    ) -> ConfigFlowResult:
         """Allow user to select device from devices connected to specified account."""
-        errors = {}
+        errors: dict[str, str] = {}
         if user_input is not None:
             device_name = user_input[DEVICE_NAME]
             system = [
                 device for device in self.fetched_systems if device.name == device_name
             ]
             device_id = system[0].id
-            await self.async_set_unique_id(device_id)
+            # This should be investigated and possibly migrated to a string
+            await self.async_set_unique_id(device_id)  # type: ignore[arg-type]
```

Comment:
Our internal type hints expect the unique_id to be a string.
Should the internal type hints be adjusted to allow integers? or should checks be added to prevent non-string values?

Reply:
This should always be a string.

Reply:
First, I would like a core member to confirm if integers should be allowed as config_entry unique_id.
It seems that we have checks to prevent it in the entity registry (eg. https://github.com/home-assistant/core/issues/116574) but I'm not sure if we have the same for the config entries.

Reply:
@mtielen based on https://github.com/home-assistant/core/pull/109966, I think that a migration is needed.
If #125602 is accepted, it will start warning in `2024.10`

This migration should be done by the code-owner, you can take a look at #124963 as an example.

Reply:
Actually the link was wrong.
I have opened a PR to migrate the `unique_id`: #125653